drivers/sx1276/sx1276.c

```diff
@@ -0,0 +1,1301 @@
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO0_00
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO2_00);
+            }
+            else {
+                sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK,  //SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                                                            //SX1276_RF_LORA_IRQFLAGS_RXDONE |
+                                                            //SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                                 SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                                 SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                                 SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                                 SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                                 SX1276_RF_LORA_IRQFLAGS_CADDETECTED);
+
+                // DIO0=RxDone
+                sx1276_reg_write(dev,
+                                 SX1276_REG_DIOMAPPING1,
+                                 (sx1276_reg_read(dev, SX1276_REG_DIOMAPPING1)
+                                  & SX1276_RF_LORA_DIOMAPPING1_DIO0_MASK)
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO0_00);
+            }
+
+            sx1276_reg_write(dev, SX1276_REG_LR_FIFORXBASEADDR, 0);
+            sx1276_reg_write(dev, SX1276_REG_LR_FIFOADDRPTR, 0);
+        }
+        break;
+    }
+
+    sx1276_set_status(dev, SX1276_RF_RX_RUNNING);
+
+    if (rx_continuous) {
+        sx1276_set_op_mode(dev, SX1276_RF_LORA_OPMODE_RECEIVER);
+    }
+    else {
+        if (timeout != 0) {
+            xtimer_set(&(dev->_internal.rx_timeout_timer), timeout);
+        }
+
+        sx1276_set_op_mode(dev, SX1276_RF_LORA_OPMODE_RECEIVER_SINGLE);
+    }
+}
+
+void sx1276_start_cad(sx1276_t *dev)
+{
+    switch (dev->settings.modem) {
+        case SX1276_MODEM_FSK:
+        {
+
+        }
+        break;
+        case SX1276_MODEM_LORA:
+        {
+            sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK, SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                             SX1276_RF_LORA_IRQFLAGS_RXDONE |
+                             SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                             SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                             SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                                                                //SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                             SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL   // |
+                                                                //SX1276_RF_LORA_IRQFLAGS_CADDETECTED
+                             );
+
+            // DIO3=CADDone
+            sx1276_reg_write(dev,
+                             SX1276_REG_DIOMAPPING1,
+                             (sx1276_reg_read(dev, SX1276_REG_DIOMAPPING1)
+                              & SX1276_RF_LORA_DIOMAPPING1_DIO0_MASK) | SX1276_RF_LORA_DIOMAPPING1_DIO0_00);
```

Comment:
Hello! Good catch. Since the CAD mode isn't tested at all, this obviously copy-paste mistake wasn't spotted before.

I'll fix that ASAP.

P. S.: the actual mistake is present in original Semtech's code:
https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1276/sx1276.c#L1164
so it was derived from there as is.

Cheers.